AutoCollection/Statsbeat.ts

```diff
@@ -0,0 +1,263 @@
+        }
+        this._lastRequests = requests;
+    }
+
+    private _trackRequestsCount() {
+        if (this._totalSuccesfulRequestCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_SUCCESS, value: this._totalSuccesfulRequestCount });
+            this._totalSuccesfulRequestCount = 0; //Reset
+        }
+        if (this._totalFailedRequestCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_FAILURE, value: this._totalFailedRequestCount });
+            this._totalFailedRequestCount = 0; //Reset
+        }
+        if (this._retryCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.RETRY_COUNT, value: this._retryCount });
+            this._retryCount = 0; //Reset
+        }
+        if (this._throttleCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.THROTTLE_COUNT, value: this._throttleCount });
+            this._throttleCount = 0; //Reset
+        }
+        if (this._exceptionCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.EXCEPTION_COUNT, value: this._exceptionCount });
+            this._exceptionCount = 0; //Reset
+        }
+    }
+
+    private async _sendStatsbeats() {
+        let envelopes: Array<Contracts.Envelope> = [];
+        let properties = {
+            "os": this._os,
+            "rp": this._resourceProvider,
+            "cikey": this._cikey,
+            "runtimeVersion": this._runtimeVersion,
+            "language": this._language,
+            "version": this._sdkVersion,
+            "attach": this._attach,
+            "instrumentation": this._instrumentations,
+            "feature": this._features,
+        }
+        for (let i = 0; i < this._statbeatMetrics.length; i++) {
+            let statsbeat: Contracts.MetricTelemetry = {
+                name: this._statbeatMetrics[i].name,
+                value: this._statbeatMetrics[i].value,
+                properties: properties
+            };
+            let envelope = EnvelopeFactory.createEnvelope(statsbeat, Contracts.TelemetryType.Metric, null, null, this._statsbeatConfig);
+            envelope.name = Constants.StatsbeatTelemetryName;
+            envelopes.push(envelope);
+        }
+        this._statbeatMetrics = [];
+        await this._sender.send(envelopes);
+    }
+
+    private _getCustomProperties() {
+        this._language = STATSBEAT_LANGUAGE;
+        this._cikey = this._config.instrumentationKey;
+        this._sdkVersion = Context.sdkVersion; // "node" or "node-nativeperf"
+        this._os = os.type();
+        this._runtimeVersion = process.version;
+    }
+
+    private _getResourceProvider(callback: () => void) {
+        // Check resource provider
+        let waiting: boolean = false;
+        if (process.env.WEBSITE_SITE_NAME) { // Web apps
```

Comment:
where is rpId? I didn't find it.

Reply:
I thought rpId only exist in "attach" Statsbeat, this PR is only adding network Statbeats will add attach and feature in different PR

Reply:
yes, rpid is part of attach statsbeat.